src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs

```diff
@@ -94,13 +96,7 @@ public void Start<TContext>(IHttpApplication<TContext> application)
                 engine.Start(threadCount);
                 var atLeastOneListener = false;
 
-                var addressesFeature = Features.Get<IServerAddressesFeature>();
-                if (addressesFeature == null)
-                {
-                    throw new InvalidOperationException($"{nameof(IServerAddressesFeature)} is missing.");
-                }
-
-                foreach (var address in addressesFeature.Addresses)
+                foreach (var address in _serverAddresses.Addresses)
```

Comment:
Would it be more correct to get the `IServerAddressesFeature` from `Features.Get()`, rather than caching it in the ctor?  Is it possible for something else to change the IServerAddressFeature between KestrelServer.ctor() and KestrelServer.Start()?